@navikt/core/tokens/darkside/figma/create-tokens.ts

```diff
@@ -0,0 +1,248 @@
+
+/**
+ * Scopes allows us to define where in Figma the token can be used.
+ * @see https://www.figma.com/plugin-docs/api/VariableScope
+ */
+function figmaSettings(token: TransformedToken): {
+  figmaType: VariableResolvedDataType;
+  scopes: VariableScope[];
+} {
+  if (isGlobalColor(token)) {
+    return createFigmaSettings("COLOR", ["ALL_FILLS", "STROKE_COLOR"]);
+  }
+  if (isBackgroundColor(token)) {
+    return createFigmaSettings("COLOR", ["FRAME_FILL", "SHAPE_FILL"]);
+  }
+  if (isBorderColor(token)) {
+    return createFigmaSettings("COLOR", ["STROKE_COLOR"]);
+  }
+  if (isContrastColor(token) || isTextColor(token)) {
+    return createFigmaSettings("COLOR", ["SHAPE_FILL", "TEXT_FILL"]);
+  }
+  if (isRadiusToken(token)) {
+    return createFigmaSettings("FLOAT", ["CORNER_RADIUS"]);
+  }
+  if (isSpacingToken(token)) {
+    return createFigmaSettings("FLOAT", ["GAP"]);
+  }
+
+  console.warn(`No fitting type or scope found for token: ${token.name}`);
+  return createFigmaSettings("STRING", []);
+}
+
+function createFigmaSettings(
+  figmaType: VariableResolvedDataType,
+  scopes: VariableScope[],
+): {
+  figmaType: VariableResolvedDataType;
+  scopes: VariableScope[];
+} {
+  return { figmaType, scopes };
+}
+
+/**
+ * Allows us to extract the token name from the token object and create the correct grouping for Figma.
+ */
+export function createTokenName(token: TransformedToken) {
+  if (!token.attributes?.item || typeof token.attributes.item !== "string") {
+    throw new Error(`No item attribute found on token: ${token.name}`);
+  }
+  /*
+   * Because of the `StyleDictionaryTokenConfig`-type, we can assume that attributes.item will always exists.
+   */
+  let name = lodash.startCase(token.attributes.item);
+
+  if (token.group) {
+    name = createGroupName(token.group) + name;
+  }
+
+  /*
+   * In the case we have a token "Text/Accent/Accent", we want to add a suffix to the name.
+   */
+  const nameKeys = new Set(name.split("/"));
+  if (nameKeys.size !== name.split("/").length) {
+    name += " Default";
+  }
+
+  /*
+   * For pure value tokens, we want to add a prefix to the name to make it more readable in Figma.
+   */
+  if (isRadiusToken(token)) {
+    return "Radius " + name;
+  }
+
+  if (isSpacingToken(token)) {
+    return "Spacing " + name;
+  }
+
+  return name;
+}
+
+/**
+ * By splitting the group by ".", we can create a more readable group name in Figma.
+ * @param group "background.primary"
+ * @returns "Background/Primary/"
+ * @see https://help.figma.com/hc/en-us/articles/14506821864087-Overview-of-variables-collections-and-modes#h_01H9V3QSVH2T1EYNXP7RNXZ8MV
+ */
+function createGroupName(group: string): string {
+  return (
+    group
+      .split(".")
+      .map((g) => lodash.startCase(g))
+      .join("/") + "/"
+  );
+}
+
+function isTokenOfSemanticColorGroup(
+  token: TransformedToken | FigmaToken,
+  groupType: keyof typeof tokenGroupLookup,
+): boolean {
+  const group = token.group as TokenGroup;
```

Comment:
This might be undefined, right?
```suggestion
  const group = token.group as TokenGroup | undefined;
```